integration-tests/src/tests/client/sharding_upgrade.rs

```diff
@@ -213,59 +192,51 @@ impl TestShardUpgradeEnv {
         // it when we are producing the block at `height`
         if let Some(txs) = self.txs_by_height.get(&(height + 1)) {
             for tx in txs {
-                let mut response_valid_count = 0;
-                let mut response_routed_count = 0;
-                for j in 0..self.num_validators {
-                    let response = env.clients[j].process_tx(tx.clone(), false, false);
-                    tracing::debug!(target: "test", client=j, tx=?tx.get_hash(), ?response, "process tx");
-                    match response {
-                        ProcessTxResponse::ValidTx => response_valid_count += 1,
-                        ProcessTxResponse::RequestRouted => response_routed_count += 1,
-                        response => {
-                            panic!("invalid tx response {response:?}");
-                        }
-                    }
-                }
-                assert_ne!(response_valid_count, 0);
-                assert_eq!(response_valid_count + response_routed_count, self.num_validators);
+                Self::process_tx(env, tx);
             }
         }
 
         // produce block
-        let block_producer = {
-            let epoch_id = env.clients[0]
-                .epoch_manager
-                .get_epoch_id_from_prev_block(&head.last_block_hash)
-                .unwrap();
-            env.clients[0].epoch_manager.get_block_producer(&epoch_id, height).unwrap()
+        let block = {
+            let client = &env.clients[0];
+            let epoch_id =
+                client.epoch_manager.get_epoch_id_from_prev_block(&head.last_block_hash).unwrap();
+            let block_producer =
+                client.epoch_manager.get_block_producer(&epoch_id, height).unwrap();
+            let _span = tracing::debug_span!(target: "test", "", client=?block_producer);
```

Comment:
nit: can directly call `.entered()` here? Can add some text to the span as well or remove altogether.